### PR TITLE
feat(haptics): adding in haptics to key areas

### PR DESCRIPTION
**Pull request by bassrock**
## Summary

Default expereinces for users on iOS these days is to expereince haptics. This adds in baseline support for some key areas of the app.

## References 
* IN-1300

## Implementation Details
* Created a haptic feedback type with @cyndichin 

## Test Steps
* Run app and click around on key actions and feel the device!

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

**Review by cyndichin — COMMENTED** [review]
some comments, but feels great!~

File: PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift [hunk 1]
```diff
@@ -293,6 +294,7 @@ class SavedItemsListViewModel: NSObject, ItemsListViewModel {
     }
 
     func overflowActions(for objectID: NSManagedObjectID) -> [ItemAction] {
+        Haptics.overflowTap()
```
[the remark this answers is not in the document]
Comment (bassrock) [reply PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift:301]: oh damn wrong one.

File: PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift [hunk 2]
```diff
@@ -318,6 +320,7 @@ class SavedItemsListViewModel: NSObject, ItemsListViewModel {
     }
 
     func trackOverflow(for objectID: NSManagedObjectID) -> UIAction? {
+        Haptics.overflowTap()
```
[the remark this answers is not in the document]
Comment (bassrock) [reply PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift:323]: @cyndichin this was the only called I could find to throw the trigger on

[the remark this answers is not in the document]
Comment (cyndichin) [reply PocketKit/Sources/PocketKit/MyList/ItemsList/SavedItemsList/SavedItemsListViewModel.swift:323]: ahh this is the SwiftUI one I was thinking, we can leave it there then! oops

**Review by cyndichin — APPROVED** [review]
thanks 🔥

**Pull request merged**